### PR TITLE
Repo Reorganization

### PR DESCRIPTION
**Pull request by sebastian-zieba**
This PR reorganizes the repository to be in the MPIA template format. The changes include:

- move _/eureka_ to _/src/eureka/_
- update paths in _setup.cfg_
- update a path in _setup.py_
- change the installation recommendations so that _pip + git_ is the first listed and add "-e" to _pip install_

### CODE REVIEW

File: docs/source/installation.rst [hunk 1]
```diff
@@ -39,7 +57,7 @@ To update your ``Eureka!`` installation to the most recent version, you can do t
 	pip install --no-deps --upgrade .
 
 
-b) With ``pip``
+c) With ``pip``
```
Comment (taylorbell57) [inline docs/source/installation.rst:60]: Can you please add the `-e` option to this installation method as well

Comment (taylorbell57) [reply docs/source/installation.rst:60]: Aside from that, as long as the automated pytest run below works successfully, I think this PR is ready to go

**Comment by sebastian-zieba** [comment]
I actually think I might have an idea what happened.. a path in the file Eureka/.github/workflows/pytest_testing.yml has to be changed. 
Instead of 
_run: pytest tests --cov=./ --cov-report=xml_
I think it should be:
_run: pytest tests --cov=./src/ --cov-report=xml_
I'm gonna try that if that's okay with you.

**Comment by kevin218** [comment]
I suggest we merge as soon as this PR is approved.  We no longer have time to wait for the NIRISS PR.

**Review by kevin218 — APPROVED** [review]

**Pull request merged**